python/ray/_private/runtime_env/working_dir.py

```diff
@@ -178,9 +182,78 @@ def modify_context(
                 "downloading or unpacking the working_dir."
             )
 
+        # Use placeholder here and will replace it by `pre_worker_startup`.
         if not _WIN32:
-            context.command_prefix += ["cd", str(local_dir), "&&"]
+            context.command_prefix += [
+                "cd",
+                WorkingDirPlugin.working_dir_placeholder,
+                "&&",
+            ]
         else:
             # Include '/d' incase temp folder is on different drive than Ray install.
-            context.command_prefix += ["cd", "/d", f"{local_dir}", "&&"]
-        set_pythonpath_in_context(python_path=str(local_dir), context=context)
+            context.command_prefix += [
+                "cd",
+                "/d",
+                WorkingDirPlugin.working_dir_placeholder,
+                "&&",
+            ]
+        set_pythonpath_in_context(
+            python_path=WorkingDirPlugin.working_dir_placeholder, context=context
+        )
+        context.cwd = WorkingDirPlugin.working_dir_placeholder
+        context.symlink_dirs_to_cwd.append(str(local_dir))
```

Comment:
Actually, why is this a list? Won't the list only ever contain one element?

Reply:
Because we could also link some contents in other plugins.